src/types.ts

```diff
@@ -1,44 +1,39 @@
-export type VoiceStartMessage = {
-  type: 'voiceStart';
+export type MeasureStartMessage = {
+  msgType: 'measureStart';
+  width: string;
+  staves: string;
+  clef: string[];
+  time: string;
 };
 
-export type VoiceEndMessage = {
-  type: 'voiceEnd';
+export type MeasureEndMessage = {
+  msgType: 'measureEnd';
 };
 
 export type BeamStartMessage = {
-  type: 'beamStart';
+  msgType: 'beamStart';
 };
 
 export type BeamEndMessage = {
-  type: 'beamEnd';
+  msgType: 'beamEnd';
 };
 
 export type NoteMessage = {
-  type: 'note';
+  msgType: 'note';
+  rest: boolean;
   stem: string;
   pitch: string;
+  type: string;
```

Comment:
In my opinion, `type` is too ambiguous for anything other than a discriminant. Can we call this `graphicNoteType` instead please? https://www.w3.org/2021/06/musicxml40/musicxml-reference/elements/type/

Reply:
type is the MusicXML naming.